internal/pkg/aws/ecs/task.go

```diff
@@ -86,12 +92,49 @@ func (t *Task) TaskStatus() (*TaskStatus, error) {
 		StoppedAt:        stoppedAt,
 		StoppedReason:    stoppedReason,
 		CapacityProvider: aws.StringValue(t.CapacityProviderName),
+		TaskDefinition:   aws.StringValue(t.TaskDefinitionArn),
 	}, nil
 }
 
 // ENI returns the network interface ID of the running task.
 // Every Fargate task is provided with an ENI by default (https://docs.aws.amazon.com/AmazonECS/latest/userguide/fargate-task-networking.html).
 func (t *Task) ENI() (string, error) {
+	attachmentENI, err := t.attachmentENI()
+	if err != nil {
+		return "", err
+	}
+
+	for _, detail := range attachmentENI.Details {
+		if aws.StringValue(detail.Name) == networkInterfaceIDKey {
+			return aws.StringValue(detail.Value), nil
+		}
+	}
+	return "", &ErrTaskENIInfoNotFound{
+		MissingField: missingFieldDetailENIID,
+		TaskARN:      aws.StringValue(t.TaskArn),
+	}
+}
+
+// PrivateIP returns the private IPv4 address of the task.
+func (t *Task) PrivateIP() (string, error) {
+	attachmentENI, err := t.attachmentENI()
+	if err != nil {
+		return "", err
+	}
+	for _, detail := range attachmentENI.Details {
+		if aws.StringValue(detail.Name) == privateIPv4AddressKey {
+			return aws.StringValue(detail.Value), nil
+		}
+	}
+	return "", &ErrTaskENIInfoNotFound{
```

Comment:
It's probably not worthy having a custom error for `privateIPv4AddressKey` currently? Since privateIPv4 address is part of the ENI information, I thought it'd appropriate to use `ErrTaskENIInfoNotFound` to cover this error, no?

Reply:
gotcha make sense